crates/router/src/core/customers.rs

```diff
@@ -37,7 +40,7 @@ pub async fn create_customer(
             .clone()
             .parse_value("AddressDetails")
             .change_context(errors::ApiErrorResponse::AddressNotFound)?;
-        db.insert_address(storage::AddressNew {
+        db.insert_address(domain::address::Address {
```

Comment:
there were 2 reasons why I went through with it.
- I actually went with using domain types till the Storage Interface traits, because they force the overall code to use domain types regardless of where ever it is being used. This forces for anything to be converted to the domain type before sending it to the storage model, making the conversion inevitable and in turn type safe
- As we had discussed that the domain types are going to be the source of truth of all types defined in the external modules, this makes the interface we defined to be plug-n-playable. If future if we ever wish to change the storage_models (highly unlikely), but this will provide a much easier time doing so, as the area for change is greatly reduced.

Reply:
Enforce the conversion in StorageInterface to use Convert